### PR TITLE
added `to_json` function

### PR DESCRIPTION
**Pull request by aznszn**
added the `to_json` function under #4216

### CODE REVIEW

**Comment by aznszn** [comment]
The last test I am a bit unsure about since trying to parse the returned into JSON into serde_json::Value results in an error

File: diesel/src/pg/expression/functions.rs [hunk 1]
```diff
@@ -1423,3 +1423,56 @@ define_sql_function! {
...
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::to_json;
+    /// #     use serde_json::Value;
+    /// #     use diesel::sql_types::{Integer, Array, Json, Text, Nullable};
+    /// #     let connection = &mut establish_connection();
+    /// let result = diesel::select(to_json::<Integer, _>(1))
+    ///     .get_result::<Value>(connection)?;
+    ///
+    /// let expected: Value = serde_json::from_str("1").unwrap();
```
Comment (weiznich) [inline diesel/src/pg/expression/functions.rs:1448]: can we use the `serde_json::json!` macro instead to construct the expected values?

File: diesel/src/pg/expression/functions.rs [hunk 2]
```diff
@@ -1423,3 +1423,56 @@ define_sql_function! {
...
+    ///
+    /// let result = diesel::select(to_json::<Nullable<Array<Text>>, _>(None::<Vec<String>>))
+    ///     .get_result::<Value>(connection);
+    ///
+    /// assert!(result.is_err());
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn to_json<E: SingleValue>(e: E) -> Json;
```
Comment (weiznich) [inline diesel/src/pg/expression/functions.rs:1477]: This function seems to return `null` if you pass in a null values. We want to have a doc test demonstrating that this is correctly propagated.

**Review by weiznich — CHANGES_REQUESTED** [review]
Thanks for working on this :pray: 

I added two comments about potential improvements, if you need help with the nullabilty handling just let me know.

File: diesel/src/pg/expression/functions.rs (hunk 1 above)
Comment (aznszn) [reply diesel/src/pg/expression/functions.rs]: done

File: diesel/src/pg/expression/functions.rs (hunk 2 above)
Comment (aznszn) [reply diesel/src/pg/expression/functions.rs:1468]: ive changed the last test a bit, let me know if its okay

**Review by weiznich — APPROVED** [review]

**Pull request merged**